File: Lab5/Lab5_CS/myFigure.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab5_CS
{
    class myFigure
    {
        protected double[,] _Coordinates;
        protected int Dots;
        public myFigure(int size, double[] XCoord, double[] YCoord, double[] ZCoord)
        {
            Dots = size;
            _Coordinates = new double[size, 3];
            for(int i = 0; i < size; i++)
            {
                _Coordinates[i, 0] = XCoord[i];
                _Coordinates[i, 1] = YCoord[i];
                _Coordinates[i, 2] = ZCoord[i];
            }
        }
        public double sideLength(int number)
        {
            double sideLength = 0;
            if (number != Dots)
            {
                for(int i = 0; i < 3; i++)
                {
                    sideLength += Math.Pow(_Coordinates[number, i] - _Coordinates[number - 1, i], 2); // Finds length of every side except for the last.
                }
                return Math.Sqrt(sideLength); 
            }
            else
            {
                for (int i = 0; i < 3; i++)
                {
                    sideLength += Math.Pow(_Coordinates[number - 1, i] - _Coordinates[0, i], 2); // Finds length of the last side.
                }
                return Math.Sqrt(sideLength);
            }
        }
        public int Len()
        {
            return Dots;
        }
    }
}

File: Lab5/Lab5_C++/myFigure.cpp
#include "myFigure.h"

myFigure::myFigure(int size, double* XCoord, double* YCoord, double* ZCoord) {
	_Dots = size;
	_Coordinates = new double* [size];
	for (auto i = 0; i < size; i++) {
		_Coordinates[i] = new double[3];
		_Coordinates[i][0] = XCoord[i];
		_Coordinates[i][1] = YCoord[i];
		_Coordinates[i][2] = ZCoord[i];
	}
}

double myFigure::SideLength(int index) {
	double sideLength = 0;
	if (index != _Dots) {
		for (int i = 0; i < 3; i++)
			sideLength += pow(_Coordinates[index][i] - _Coordinates[index - 1][i], 2);
		return sqrt(sideLength);
	}
	else {
		for (int i = 0; i < 3; i++)
			sideLength += pow(_Coordinates[index][i] - _Coordinates[0][i], 2);
		return sqrt(sideLength);
	}
}

int myFigure::Dots() {
	return _Dots;
}
